lua/projectlaunch/runtime_variable_manager.lua

```diff
@@ -0,0 +1,42 @@
+local RuntimeVarManager = {}
+local vars = {}
+
+local function createArgSuffix()
+	return " (" .. table.concat(vars, ",") .. ")"
+end
+
+local function setRuntimeVarViaPrompt(i, command)
+	local substitutionString = "$" .. i
+	vim.ui.input(
+		{ prompt = "ProjectLaunch: Enter argument " .. substitutionString .. " for\n" .. command .. "\n:" },
+		function(input)
+			vars[i] = input
+		end
+	)
+end
+
+function RuntimeVarManager.generateJobName(command)
+	if vim.tbl_count(vars) > 0 then
+		return command.name .. createArgSuffix()
+	end
+
+	return command.name
+end
+
+function RuntimeVarManager.interpolate(originalCommand)
+    vars = {}
```

Comment:
My one last concern is with this `vars` variable still. I'd rather not have this `vars` variable work like this where subsequent calls (like to generateJobName) rely on having called interpolate first for `vars` to be relevant. Maybe `interpolate` could return both the name and the command in a table? Like since we're still passing `command` to both, it seems unintuitive to me that generateJobName implicitly depends on interpolate to have been called with the same command first. I don't see any obvious bugs with this, but if things were different I could see running into issues.

Also, I think it's totally fine to use the final command with substitutions as the name! What was your motivation for doing it this way?

Let me know if this makes sense at all!